### PR TITLE
Add a set of search service passthrough endpoints

### PR DESCRIPTION
**Pull request by ianmcorvidae**
(for data search only for now, of course)

This should be pretty straightforward? But, I want to make sure I'm still using best-practices since I haven't been doing much clojure lately.

### CODE REVIEW

**Comment by ianmcorvidae** [comment]
Oh, and I haven't documented this yet, but that goes in other repos anyway.

**Comment by slr71** [comment]
Looks good to me. :shipit:

**Comment by ianmcorvidae** [comment]
Cool, thanks for the review!

File: src/terrain/clients/search.clj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+(ns terrain.clients.search
+  (:use [terrain.util.transformers :only [secured-params]]
```
Comment (psarando) [inline src/terrain/clients/search.clj]: It doesn't look like `secured-params` is used here yet (but we can leave it for now if we're planning on passing the user params to the search service in the future).

File: src/terrain/util/config.clj [hunk 2]
```diff
@@ -504,6 +504,18 @@
...
+(cc/defprop-optstr search-base-url
+  "The hostname of the search service"
+  [props config-valid configs]
+  "terrain.search.base-url" "http://search:60000")
+
+(def search-base
+  (memoize
+   (fn []
+     (if (System/getenv "SEARCH_PORT")
+       (cfg/env-setting "SEARCH_PORT")
```
Comment (psarando) [inline src/terrain/util/config.clj]: Should we name this `SEARCH_URL` or something? The `else` is returning `search-base-url`...

File: src/terrain/clients/search.clj (hunk 1 above)
Comment (ianmcorvidae) [reply src/terrain/clients/search.clj]: oh, you're right. I originally used it, but the structuring using `terrain.util/controller` means I didn't need to, but I forgot to remove the import.

File: src/terrain/util/config.clj (hunk 2 above)
Comment (ianmcorvidae) [reply src/terrain/util/config.clj]: hm, I think we don't use the environment-based configs anyway; I was copying the others which have the same weirdness. I think I'll just remove this and use `search-base-url` directly

**Pull request merged**